collectors/flaw_labels/core.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import requests
+import yaml
+from django.conf import settings
+from requests_gssapi import HTTPSPNEGOAuth
+
+from osidb.helpers import ensure_list, get_model_fields
+from osidb.models import FlawLabel
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_flaw_labels(url):
+    """
+    Fetch Flaw Labels from given url.
+    """
+    response = requests.get(
+        url=url,
+        auth=HTTPSPNEGOAuth(),
+        timeout=settings.DEFAULT_REQUEST_TIMEOUT,
+    )
+    response.raise_for_status()
+
+    try:
+        labels = yaml.safe_load(response.text)
+        return (labels["context_based_labels"], labels["product_family_labels"])
+    except yaml.YAMLError as e:
+        logger.error("Error parsing YAML", exc_info=e)
+
+
+def sync_flaw_labels(context_based: dict, product_family: dict):
```

Comment:
This should probably be atomic so it cannot happen that we delete the labels and then encounter an error interrupting the creation resulting in having no labels at all.

Reply:
done!